src/utils/notification-utils.spec.js

```diff
@@ -192,4 +192,133 @@ describe('notification-utils', () => {
     });
   });
 
+  describe('groupNotifications', () => {
+    let result;
+    let notifications;
+
+    describe('when all notifications are considered distinct', () => {
+      beforeEach(() => {
+        notifications = [
+          { _id: 'notification-1', eventType: EVENT_TYPE.revisionCreated, eventParams: { documentId: 'document-1' }, createdOn: new Date('2023-02-28T12:01:00Z') },
+          { _id: 'notification-2', eventType: EVENT_TYPE.revisionCreated, eventParams: { documentId: 'document-2' }, createdOn: new Date('2023-02-28T12:02:00Z') },
+          { _id: 'notification-3', eventType: EVENT_TYPE.revisionCreated, eventParams: { documentId: 'document-3' }, createdOn: new Date('2023-02-28T12:03:00Z') }
+        ];
+
+        result = groupNotifications(notifications);
+      });
+
+      it('should create a new group for each notification', () => {
+        expect(result).toStrictEqual([
+          {
+            notificationIds: ['notification-1'],
+            eventType: EVENT_TYPE.revisionCreated,
+            eventParams: { documentId: 'document-1' },
+            firstCreatedOn: new Date('2023-02-28T12:01:00Z'),
+            lastCreatedOn: new Date('2023-02-28T12:01:00Z')
+          },
+          {
+            notificationIds: ['notification-2'],
+            eventType: EVENT_TYPE.revisionCreated,
+            eventParams: { documentId: 'document-2' },
+            firstCreatedOn: new Date('2023-02-28T12:02:00Z'),
+            lastCreatedOn: new Date('2023-02-28T12:02:00Z')
+          },
+          {
+            notificationIds: ['notification-3'],
+            eventType: EVENT_TYPE.revisionCreated,
+            eventParams: { documentId: 'document-3' },
+            firstCreatedOn: new Date('2023-02-28T12:03:00Z'),
+            lastCreatedOn: new Date('2023-02-28T12:03:00Z')
+          }
+        ]);
+      });
+    });
+
+    describe('when there are contiguous notifications of the same type relating to the same document', () => {
```

Comment:
Now **that's** a word for ya! Contiguous..

Reply:
Finally, haha :laughing: